CondFormats/PPSObjects/src/PPSPixelTopology.cc

```diff
@@ -28,16 +28,17 @@ unsigned short PPSPixelTopology::pixelIndex(PixelInfo pI) const {
 
 bool PPSPixelTopology::isPixelHit(float xLocalCoordinate, float yLocalCoordinate, bool is3x2 = true) const {
   // check hit fiducial boundaries
-  double xModuleSize = 2 * ((no_of_pixels_simX_ / 2. + 1) * pitch_simX_ + dead_edge_width_);
+  const double xModuleSize = 2 * ((no_of_pixels_simX_ / 2. + 1) * pitch_simX_ + dead_edge_width_);
   if (xLocalCoordinate < -xModuleSize / 2. || xLocalCoordinate > xModuleSize / 2.)
     return false;
 
-  double yModuleSize = (no_of_pixels_simY_ + 4.) * pitch_simY_ + 2. * dead_edge_width_;
-  double y2x2top = no_of_pixels_simY_ / 6. * pitch_simY_ + dead_edge_width_;
+  const double yModuleSize = (no_of_pixels_simY_ + 4.) * pitch_simY_ + 2. * dead_edge_width_;
+  const double y2x2top = no_of_pixels_simY_ / 6. * pitch_simY_ + dead_edge_width_;
   if (is3x2 && (yLocalCoordinate < -yModuleSize / 2. || yLocalCoordinate > yModuleSize / 2.))
     return false;
-
-  if (!is3x2 && (yLocalCoordinate < -yModuleSize / 2. || yLocalCoordinate > y2x2top))
+  if (!is3x2 && (runType_ == "Run2") && (yLocalCoordinate < -yModuleSize / 2. || yLocalCoordinate > y2x2top))
```

Comment:
Can `runType_` be different from "Run2" and  "Run3"? What happens in that case?
Shouldn't a default value be allowed?

Reply:
While @fabferro can possibly comment further, my personal point of view is the following.

`Run2` and `Run3` are, conceptually, the only legitimate values - PPS was installed during Run2 and now we are preparing for Run3. Technically, the class now allows for other values, too, so it is upon the expert who prepares the DB payloads to ensure the data integrity. If you deem that a check is desirable, then I believe we can add one. In this case I could think of a check in the constructor in this class or a check in the associated ESProducer - would you have a preference?

Regarding the current implementation, for a configuration which is not checked, the method returns `true`, i.e. the hit is in sensor acceptance. It might be possible better/safer to return `false` for an unknown/illegal configuration - in the sense that the sensor has no acceptance except for the known and well defined cases. What do you think?